$OpenBSD: patch-gdk-pixbuf_gdk-pixbuf-io_c,v 1.9 2008/12/20 19:37:19 ajacoutot Exp $

Lower LOAD_BUFFER_SIZE so that we don't run into memory access issues.
See GNOME bugzilla #561938.

--- gdk-pixbuf/gdk-pixbuf-io.c.orig	Mon Nov 24 07:35:45 2008
+++ gdk-pixbuf/gdk-pixbuf-io.c	Sat Dec 20 17:32:11 2008
@@ -49,7 +49,7 @@
 #endif
 
 #define SNIFF_BUFFER_SIZE 4096
-#define LOAD_BUFFER_SIZE 65536
+#define LOAD_BUFFER_SIZE 4096
 
 #ifndef GDK_PIXBUF_USE_GIO_MIME 
 static gint 
@@ -298,7 +298,7 @@ gdk_pixbuf_get_module_file (void)
   gchar *result = g_strdup (g_getenv ("GDK_PIXBUF_MODULE_FILE"));
 
   if (!result)
-	  result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gdk-pixbuf.loaders", NULL);
+	  result = g_build_filename ("/var/db", "gtk-2.0", "gdk-pixbuf.loaders", NULL);
 
   return result;
 }
